python/lsst/pipe/base/configOverrides.py

```diff
@@ -25,15 +25,108 @@
 __all__ = ["ConfigOverrides"]
 
 import ast
+from operator import attrgetter
 
 import lsst.pex.exceptions as pexExceptions
 from lsst.utils import doImport
 
+import inspect
 from enum import Enum
 
 OverrideTypes = Enum("OverrideTypes", "Value File Python Instrument")
 
 
+class ConfigExpressionParser(ast.NodeVisitor):
+    """An expression parser that will be used to transform configuration
+    strings supplied from the command line or a pipeline into a python
+    object.
+
+    This is roughly equivalent to ast.literal_parser, but with the ability to
+    transform strings that are valid variable names into the value associated
+    with the name. Variables that should be considered valid are supplied to
+    the constructor as a dictionary that maps a string to its corresponding
+    value.
+
+    This class in an internal implementation detail, and should not be exposed
+    outside this module.
+
+    Parameters
+    ----------
+    namespace : `dict` of `str` to variable
+    """
+
+    def __init__(self, namespace):
+        self.variables = namespace
+
+    def visit_Name(self, node):
+        """This method gets called when the parser has determined a node
+        corresponds to a variable name.
+        """
+        # If the id (name) of the variable is in the dictionary of valid names,
+        # load and return the corresponding variable.
+        if node.id in self.variables:
+            return self.variables[node.id]
+        # If the node does not correspond to a valid variable, turn the name
+        # into a string, as the user likely intended it as such.
+        return f"{node.id}"
+
+    def visit_List(self, node):
+        """This method is visited if the node is a list. Constructs a list out
+        of the sub nodes.
+        """
+        result = []
+        for elm in node.elts:
+            result.append(self.visit(elm))
+        return result
```

Comment:
`return [self.visit(elm) for elm in node.elts]` should be shorter?

Reply:
And again, I know nothing about `ast` but would it be possible to skip implementing `visit_List` if you inherit from `NodeTransformer` instead of `NodeVisitor`?

Reply:
`NodeTransformer` is designed to transform an `ast`, into another `ast`, that will later be consumed by some other visitor system. In our case we actually want to take in the `ast` and emit actual python processing.

The methods that were implemented were chosen to represent the range of things we expect to handle processing snippets from pipelines etc. The advantage of this approach is that we can expand to handle more specific syntax in the future if we choose to.